### ISSUE TITLE
Form Plugin update overwrite the user preferences 

### ISSUE THREAD

**Issue by pmoreno-rodriguez** [getgrav/grav#3194]
Hi.
In the latest form plugin updates, I've noticed that this overwrite the user preferences. In my case, I have recaptcha with V3 - Latest version.

In each update of the plugin this value returns to version 2, making the form unusable.

Is there any way to correct this issue?
Thanks so much.

**Comment by rhukster** [comment]
You mean when you save things in admin, it doesn't save your changes? Or do you mean when you updated you lost your configuration?  If the second, did you make changes in `user/plugins/form/form.yaml` ? If so that's normal that an update would overwrite those changes.  Changes should not be made in this location, should be made in `user/config/plugins/form.yaml` per the README/Grav docs etc.

**Comment by pmoreno-rodriguez** [comment]
Hi @rhukster 
I have my form.yaml file in user/config/plugins with this content:
```
enabled: true
built_in_css: true
inline_css: true
refresh_prevention: false
client_side_validation: true
inline_errors: false
files:
  multiple: false
  limit: 10
  destination: self@
  avoid_overwriting: false
  random_name: false
  filesize: 0
  accept:
    - 'image/*'
recaptcha:
  version: '3'
  theme: light
  site_key: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
  secret_key: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```
The original form.yaml in user/plugin/form has the following:
```
enabled: true
built_in_css: true
inline_css: true
refresh_prevention: false
client_side_validation: true
inline_errors: false
files:
  multiple: false          # To allow multiple files, default is single
  limit: 10                # Number of allowed files per field (multiple required)
  destination: 'self@'     # Where to upload the files (path and self@, page@, theme@)
  avoid_overwriting: false # Prevent files with the same name to be overridden. Date prefix will be added
  random_name: false       # Generate a random 15 long string name for the uploaded files
  filesize: 0              # Maximum file size allowed (in MB)
  accept:                  # List of mime/types or file extensions allowed (ie, image/*,.zip,.mp4)
    - image/*
recaptcha:
    version: 2-checkbox
    theme: light
    site_key:
    secret_key:
```
When I update the form plugin (it has happened in two or three last updates), the recaptcha version changes to **2-checkbox**. However, both site_key and secret_key keep their values. I don't know why this happens.

**Comment by rhukster** [comment]
i'll have to dig into this.

**Comment by Karmalakas** [comment]
It's not just Form plugin. Almost every update messes with users configs most of the time. The most noticeable is when translations turn off and Admin shows labels instead of translated strings. I haven't encountered this one, but latest Grav upgrade 1.7.3 >> 1.7.5 and all available plugins updates simply removed all related custom configs from _/user/env/mypage.local/config/..._. And not only under _/plugins_, but also _media.yaml_, _security.yaml_, _system_, etc.

**Comment by rhukster** [comment]
There's really no logic to change user configurations in the update code of Grav.

I tested your original setup, with your `/user/config/plugins/form.yaml` configuration with a previous version of form installed.  Then I updated the Form plugin and the user config was left unchanged.

The same goes for any plugin really, all the update code does is download the new file, remove the old `user/plugins/THE_PLUGIN` folder and then extract the new version back into the `user/plugins/THE_PLUGIN` folder.  

This location of course is whipped out so any changes in the `user/plugins` folder would be lost. However, nothing is touched in `user/config/plugins/` that is where your overrides and changes are stored, and they are never touched.

The only suggestion I have is to review your plugins, specifically 3rd party non-Team Grav or Trilby plugins, as we cannot attest to what those are doing.

**Comment by mahagr** [comment]
Grav packages don't even have `user/env` folder and even if they did, `user` is ignored while copying the files. Plugin updates only replace a single plugin folder. So it should really be impossible for configuration to be replaced (and none of us have ever seen this issue). 

I cannot see anything lese but 3rd party plugin doing this (and yes, I've just gone through the whole installation logic) a week ago.

**Comment by w00fz** [comment]
@pmoreno-rodriguez , @Karmalakas I wonder if by any chance you guys both happen to share a plugin that is causing this? Would be great to identify what could that be so we can intervene by fixing it or informing the owner about the issue.

Do you guys mind both listing all the plugins under your `user/plugins` folder?

Thanks,

**Comment by Karmalakas** [comment]
Here are my plugins. Media metadata and Cookie consent weren't updated since I started using Grav
![image](https://user-images.githubusercontent.com/3602819/106790877-dadef000-665c-11eb-9f59-43384ef8ddaf.png)

**Comment by w00fz** [comment]
Did you have `Admin Addon Media Metadata` installed when you updated and the issue happened? I think it could be that plugin.

**Comment by Karmalakas** [comment]
I had, but I've also updated from RC to 1.7.0 to 1.7.3 with this plugin installed and had no issues. Update to 1.7.5 and all plugins that had update available (my bad I didn't do separate updates and right after Grav updated I ran all plugins update) removed custom configs. There were a few posts on forum also, when after update translations were switched off. I guess it was same result - custom config missing.

**Comment by rhukster** [comment]
Please try to recreate with those two non-green-checked plugins disabled.

**Comment by Karmalakas** [comment]
That might take some time to reproduce. I'd have to get some previous version and try, but then again, as I said, it seems like a random thing. At least from my experience and from what I saw on forum. It should be a bit difficult to reproduce the issue to not happen when usually it doesn't happen :D Will see what I can do, but now I have other Grav issues to solve, that are more important to me :)

**Comment by pmoreno-rodriguez** [comment]
Here are my plugins. I'm using Grav 1.6.31, not 1.7. Sorry for screenshot quality. It's by mobile. If you can't view clearly, I'll try to upload another with better quality.

![IMG_20210203_212120](https://user-images.githubusercontent.com/60596353/106804725-1205cf00-6666-11eb-809b-67f5e905d5b1.jpg)

**Comment by pmoreno-rodriguez** [comment]
Hi, again.

I just to updated an older local site with Grav 1.6.31 and the Form, Email and Login plugins out of date. Before that, I've changed the recaptcha version in form plugin, and then it creates the form.yaml in /user/config/plugins with the new configuration.

Now, I've not been able to recreate the issue. All works fine. Maybe the website in which I detected this problem has some plugin that changes form.yaml options. I don't know and I'll keep watching this.

Thanks so much for your help.
